virtcontainers/qemu.go

```diff
@@ -663,6 +663,7 @@ func (q *qemu) setupVirtiofsd() (err error) {
 	if err != nil {
 		return err
 	}
+	defer fd.Close()
```

Comment:
Shouldn't we remove the `fd.Close()` call a few lines below?

Reply:
OOI, I had wondered if we needed to drop the later `fd.Close()` as well - but, hmm, as it is it means we close the `fd` after its last use. If we drop that, it will stay open for longer when not needed. On the flip side, we can only leave it there if calling `fd.Close()` on an already closed `fd` is safe - as it will still be called by the `defer`...

Reply:
The intent of that `fd.Close()` is clearly to close the `fd` before entering the loop, so I left it there intentionally, after checking that closing twice was OK.

Reply:
So, just to confirm, the final action on that is to remove the fd.Close() line mentioned, considering "**Close will return an error if it has already been called**". is it?

(Christophe's Emphasis ;-))

Reply:
Ah, I see @c3d's point after a private conversation.

fd.Close() will error out with "already closed", but won't crash, so that may be fine as we don't check / report that error.

I'm fine with this as is now. @jcvenegas, do you agree? If so, feel free to remove the "do-not-merge" label and merge it.

Reply:
Hey thanks for the clarifications, I agree that will not be harmful in the case if it is called two times.  I think is just a matter of different styles but the same result. let's merge it :)